mlflow/models/dependencies_schemas.py

```diff
@@ -22,10 +23,12 @@ class DependenciesSchemasType(Enum):
 
 @experimental
 def set_retriever_schema(
+    *,
     primary_key: str,
     text_column: str,
     doc_uri: Optional[str] = None,
     other_columns: Optional[List[str]] = None,
+    name: Optional[str] = "retriever",
```

Comment:
```suggestion
    name: str = "retriever",
```

Do we need `Optional`?

Reply:
It is an optional argument with `"retriever"` as default value

Reply:
not a blocker at all but https://docs.python.org/3/library/typing.html#typing.Optional

<img width="831" alt="image" src="https://github.com/mlflow/mlflow/assets/17039389/ac2dc95b-a8a7-46aa-a85d-d5ed7d14fc88">

Reply:
Never mind this comment. Using `Optional` here is gonna break nothing.